packages/core/src/operations.ts

```diff
@@ -228,7 +228,11 @@ export class TxOperations implements Omit<Client, 'notify'> {
   ): Promise<TxResult> {
     const hierarchy = this.client.getHierarchy()
     const mixClass = Hierarchy.mixinOrClass(doc)
-    if (hierarchy.isMixin(mixClass)) {
+    const updateKeys = Object.keys(update).filter((key) => !key.startsWith('$'))
```

Comment:
You need to take all attributes and go one by one and put into different parent mixins and original parent document, so it could be P? + n*M? transactions for every mixin in chain and parent document in case it has attribute changes for proper easy to use implementation.